OneSignalSDK/onesignal/src/main/java/com/onesignal/OSInAppMessageController.java

```diff
@@ -94,6 +106,22 @@ protected OSInAppMessageController() {
         );
         if (tempClickedMessageIdsSet != null)
             clickedClickIds.addAll(tempClickedMessageIdsSet);
+
+        initRedisplayData(dbInstance);
+    }
+
+    /*
+    * For testing purposes
+    * */
+    public DateGenerator getDateGenerator() {
```

Comment:
We can clean up this `DateGenerator` by using  `System.currentTimeMillis()`  instead.  This way we can use `advanceSystemTimeBy` in our tests for mocking time.

Reply:
oh men, why i didn't see this